hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java

```diff
@@ -608,24 +629,26 @@ private List<StateChanges.Builder> onMigrate(
             final var config = configProvider.getConfiguration();
             final var ledgerConfig = config.getConfigData(LedgerConfig.class);
             final var genesisRoster = createRoster(requireNonNull(genesisAddressBook));
-
             genesisNetworkInfo = new GenesisNetworkInfo(genesisRoster, ledgerConfig.id());
         }
-        final List<StateChanges.Builder> migrationStateChanges = new ArrayList<>();
-        // (FUTURE) In principle, the FileService could actually change the active configuration during a
-        // migration, which implies we should be passing the config provider and not a static configuration
-        // here; but this is a currently unneeded affordance
         blockStreamService.resetMigratedLastBlockHash();
+        startupNetworks = startupNetworksFactory.apply(selfNodeId.id(), configProvider, tssBaseService);
+        PLATFORM_STATE_SERVICE.setActiveRosterFn(
+                () -> new ReadableRosterStoreImpl(state.getReadableStates(RosterService.NAME)).getActiveRoster());
         final var migrationChanges = serviceMigrator.doMigrations(
                 state,
                 servicesRegistry,
                 deserializedVersion,
                 version,
+                // (FUTURE) In principle, the FileService could change the active configuration during a
+                // migration, implying we should pass a config provider; but we don't need this yet
                 configProvider.getConfiguration(),
                 platformConfiguration,
                 genesisNetworkInfo,
-                metrics);
-        migrationStateChanges.addAll(migrationChanges);
+                metrics,
+                startupNetworks);
+        PLATFORM_STATE_SERVICE.clearActiveRosterFn();
```

Comment:
Why do we need to clear it here ?

Reply:
I wanted to do this just to emphasize the scope of this `Roster` supplier is very limited---only to the code running inside `Schema`s.